### PR TITLE
replace string functions with multi byte forms

### PR DESCRIPTION
**Pull request by mehdihasanpour**

### CODE REVIEW

**Comment by nbayramberdiyev** [comment]
This branch is outdated.

**Comment by mehdihasanpour** [comment]
> This branch is outdated.

It is updated right now.

File: Slim/App.php [hunk 1]
```diff
@@ -205,7 +205,7 @@ public function handle(ServerRequestInterface $request): ResponseInterface
          * routing logic which could potentially append content to the response body
          * https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.4
          */
-        $method = strtoupper($request->getMethod());
+        $method = mb_strtoupper($request->getMethod());
```
Comment (nbayramberdiyev) [inline Slim/App.php:208]: Do we really need this since we're not operating on a Unicode string?

File: Slim/Middleware/BodyParsingMiddleware.php [hunk 2]
```diff
@@ -175,7 +175,7 @@ protected function getMediaType(ServerRequestInterface $request): ?string
 
         if (is_string($contentType) && trim($contentType) !== '') {
             $contentTypeParts = explode(';', $contentType);
-            return strtolower(trim($contentTypeParts[0]));
+            return mb_strtolower(trim($contentTypeParts[0]));
```
Comment (nbayramberdiyev) [inline Slim/Middleware/BodyParsingMiddleware.php:178]: Same here.

File: Slim/Middleware/MethodOverrideMiddleware.php [hunk 3]
```diff
@@ -26,7 +26,7 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
 
         if ($methodHeader) {
             $request = $request->withMethod($methodHeader);
-        } elseif (strtoupper($request->getMethod()) === 'POST') {
+        } elseif (mb_strtoupper($request->getMethod()) === 'POST') {
```
Comment (nbayramberdiyev) [inline Slim/Middleware/MethodOverrideMiddleware.php:29]: Same here.

File: Slim/ResponseEmitter.php [hunk 4]
```diff
@@ -58,7 +58,7 @@ public function emit(ResponseInterface $response): void
     private function emitHeaders(ResponseInterface $response): void
     {
         foreach ($response->getHeaders() as $name => $values) {
-            $first = strtolower($name) !== 'set-cookie';
+            $first = mb_strtolower($name) !== 'set-cookie';
```
Comment (nbayramberdiyev) [inline Slim/ResponseEmitter.php:61]: Same here.

File: tests/AppTest.php [hunk 5]
```diff
@@ -194,7 +194,7 @@ public function testGetPostPutPatchDeleteOptionsMethods(string $method): void
             return $this;
         });
 
-        $methodName = strtolower($method);
+        $methodName = mb_strtolower($method);
```
Comment (nbayramberdiyev) [inline tests/AppTest.php:197]: Same here.

**Review by nbayramberdiyev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**